### PR TITLE
formulae across sheets, selection fixes and autofill fixes

### PR DESCRIPTION
**Pull request by aleschneck**
- other sheets can be referenced in [SheetX!] in formulae
- autofill fixes
- selection fixes

### CODE REVIEW

**Review by StefanRoggenstein — APPROVED** [review]

File: src/component/selector.js [hunk 1]
```diff
@@ -320,6 +320,12 @@ export default class Selector {
     setAllAreaOffset.call(this, this.data.getSelectedRect());
   }
 
+  setStartEnd(sri, sci, eri, eci, mri, mci) {
```
Comment (colourgarden) [inline src/component/selector.js:324]: @aleschneck Can these variables be extended to give some meaning? I have no idea what `sri`, `eci` etc means. Or a brief comment to explain

File: src/component/sheet.js [hunk 2]
```diff
@@ -392,6 +392,7 @@ function overlayerMousedown(evt) {
   // sort or filter
   const { autoFilter } = data;
   if (autoFilter.includes(ri, ci)) {
+    // console.log('autoFilter.includes', { ri, ci });
```
Comment (colourgarden) [inline src/component/sheet.js]:
```suggestion
```

File: src/component/sheet.js [hunk 3]
```diff
@@ -915,14 +934,15 @@ function sheetInitEvents() {
 }
 
 export default class Sheet {
-  constructor(targetEl, data) {
+  constructor(targetEl, data, datas) {
```
Comment (colourgarden) [inline src/component/sheet.js:938]: @aleschneck What is the use of the difference between `data` and `datas`? I think we may benefit from a quick comment to explain here.

File: src/component/table.js [hunk 4]
```diff
@@ -300,14 +307,16 @@ function renderFreezeHighlightLine(fw, fh, ftw, fth) {
 
 /** end */
 class Table {
-  constructor(el, data) {
+  constructor(el, data, datas) {
```
Comment (colourgarden) [inline src/component/table.js:312]: @aleschneck Ditto `datas`

File: src/core/row.js [hunk 5]
```diff
@@ -111,6 +111,7 @@ class Rows {
 
   // what: all | format | text
   copyPaste(srcCellRange, dstCellRange, what, autofill = false, cb = () => {}) {
+    console.log('row.copyPaste')
```
Comment (colourgarden) [inline src/core/row.js]:
```suggestion
```

File: src/core/data_proxy.js [hunk 6]
```diff
@@ -487,11 +487,34 @@ export default class DataProxy {
...
-  autofill(cellRange, what, error = () => {}) {
-    const srcRange = this.selector.range;
-    if (!canPaste.call(this, srcRange, cellRange, error)) return false;
+  autofill(dpSelector, what, error = () => {}) {
+    const { range } = this.selector;
+    if (!canPaste.call(this, range, dpSelector.arange, error)) return false;
     this.changeData(() => {
-      copyPaste.call(this, srcRange, cellRange, what, true);
+      const {
+        sri: srcSri, sci: srcSci, eri: srcEri, eci: srcEci,
```
Comment (colourgarden) [inline src/core/data_proxy.js:495]: @aleschneck Same here with variable names

**Review by colourgarden — APPROVED** [review]
Approving as I don't see any major issues but please take a look at the comments.

General point - please don't shorten function or variable names just to save a few characters - we're not on a budget 😉 

`sheetRegex` rather than `sheetRgx`, for example.

File: src/component/sheet.js (hunk 3 above)
Comment (aleschneck) [reply src/component/sheet.js:938]: this is why typescript was invented :)

File: src/component/selector.js (hunk 1 above)
Comment (aleschneck) [reply src/component/selector.js:324]: indeed they can, the thing is though: you'll find these variables throughout the entirety of the code, I don't like it either but didn't want to be inconsistent, s = start, e = end, m = move, r = row, c = column, i = index, thus sri = startRowIndex

**Pull request merged**